ai_ta_backend/filtering_contexts.py

```diff
@@ -73,7 +75,7 @@ def filter_context(context, user_query, langsmith_prompt_obj):
           max_tokens=250,
       )
     completion = ret["choices"][0]["message"]["content"]
-    print("API call time: ", (time.monotonic() - start_time))
+    print("API call time: ", (time.monotonic() - api_start_time))
```

Comment:
@KastanDay I am logging the time taken for openAI response here and I am getting the below results on Railway. There is a significant time difference in the last few calls (7s to 131s). Any idea why that could be? 
<img width="395" alt="image" src="https://github.com/UIUC-Chatbot/ai-ta-backend/assets/25609228/c60705c8-af03-42ca-9947-52fde12170fe">
These are the results on my local for comparison: 
<img width="267" alt="image" src="https://github.com/UIUC-Chatbot/ai-ta-backend/assets/25609228/00e9dfdc-0da4-41ef-8704-40a51541ce8a">

Reply:
> @KastanDay I am logging the time taken for anyscale response here and I am getting the below results on Railway. There is a significant time difference in the last few calls (7s to 131s). Any idea why that could be? <img alt="image" width="395" src="https://private-user-images.githubusercontent.com/25609228/289759538-c60705c8-af03-42ca-9947-52fde12170fe.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDIzNjEzMzQsIm5iZiI6MTcwMjM2MTAzNCwicGF0aCI6Ii8yNTYwOTIyOC8yODk3NTk1MzgtYzYwNzA1YzgtYWYwMy00MmNhLTk5NDctNTJmZGUxMjE3MGZlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjEyVDA2MDM1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTEzYWVkYmUzYTc3NjM5ZTUyN2EzNGFmMTE4ZDI0NDZlOTg3ZWI1MmNkZmVlMDNhMDU2M2MzYTU1MDgyY2Q4NDgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.tjfUcSre3qUrVAuV9X9nN4HbEv9wmrdOPyvy7gHCES4"> These are the results on my local for comparison: <img alt="image" width="267" src="https://private-user-images.githubusercontent.com/25609228/289759758-00e9dfdc-0da4-41ef-8704-40a51541ce8a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDIzNjEzMzQsIm5iZiI6MTcwMjM2MTAzNCwicGF0aCI6Ii8yNTYwOTIyOC8yODk3NTk3NTgtMDBlOWRmZGMtMGRhNC00MWVmLTg3MDQtNDBhNTE1NDFjZThhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjEyVDA2MDM1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI2NWIwYWRmZjVlOTVjNDUxN2IyZWFhNWYwYmM4ZTVmZjExMDU2YmRjY2Q4ZmZlZjU0NjE4MDg5ZWMwYTllOTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.pkJl_AL3ztLhZy4LUToySyszmCVMYbzB_VPj2GSqYWM">